src/config/args.rs

```diff
@@ -88,15 +91,16 @@ impl RawBaseArgs {
     /// it is assumed to be a sibling to the config file.
     /// Database implementations then create it if it does not exist.
     pub async fn validate(self) -> Result<BaseArgs, Report<Error>> {
-        // Need to set data root first, since other things in this function use it.
-        if let Some(data_root) = self.data_root {
-            set_data_root(data_root).change_context(Error::SetDataRoot)?;
-        }
+        let ctx = AppContext::new(if let Some(data_root) = self.data_root {
```

Comment:
I think this would be more readable if you created a `data_root` variable here. Up to you, though!

```rust
let data_root = self.data_root.unwrap_or(default_data_root().await?);
let ctx = AppContext::new(data_root);
```